### PR TITLE
Tools

### PR DESCRIPTION
**Pull request by MGatner**
* Updates the development toolkit
* Fixes and optimizes test cases
* Fixes and applies static analysis changes

Note: There are a number of files not PSR-12 compliant, but I have not run them through the style fixer yet to keep this PR clean.

### CODE REVIEW

File: depfile.yaml [hunk 1]
```diff
@@ -0,0 +1,153 @@
+paths:
+  - ./src
+  - ./vendor/codeigniter4/codeigniter4/system
+  - ./vendor/tatter
```
Comment (lonnieezell) [inline depfile.yaml:4]: I don't believe we're pulling in any `tatter` dependencies.

**Review by lonnieezell — APPROVED** [review]
I think we have as many files dedicated to third-party tools as we do actual code now. Can't decide if that's good or bad lol.

File: depfile.yaml (hunk 1 above)
Comment (MGatner) [reply depfile.yaml]: Yep, templates got me 😂 Fix should already be in place.

**Comment by MGatner** [comment]
I know it's a pain, but they end up providing a lot of invaluable automation for the process and can save a lot of redoing things down the road. I work very hard at having a "set-and-forget" workflow in place so it doesn't feel like the tools take over; the setting up part is messy though.

**Comment by MGatner** [comment]
I just activated this repo in Coveralls - probably takes it a little while to catch up so that failure is not a problem.

**Pull request merged**